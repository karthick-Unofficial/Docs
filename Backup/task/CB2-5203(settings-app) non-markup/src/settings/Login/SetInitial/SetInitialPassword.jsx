import React, { Component } from "react";

import TextField from "material-ui/TextField";
import RaisedButton from "material-ui/RaisedButton";
import Divider from "material-ui/Divider";

import { browserHistory } from "react-router";

import { authService } from "client-app-core";
import { Translate, getTranslation } from "orion-components/i18n/I18nContainer";

export default class SetInitialPassword extends Component {
    constructor(props) {
        super(props);

        this.state = {
            passwordValue: "",
            passwordConfirmValue: "",
            errorMessage: this.props.errorMessage || null,
            verified: false,
            done: false
        };
    }

    componentDidMount() {
        authService.verifySetInitialPassword(
            this.props.params.token,
            (err, response) => {
                if (err) {
                    console.log(err);
                    browserHistory.push("/login/invalid-reset");
                } else {
                    this.setState({
                        verified: true
                    });
                }
            }
        );
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.errorMessage) {
            this.setState({
                errorMessage: nextProps.errorMessage
            });
        }
    }

    handlePasswordChange = e => {
        this.setState({
            passwordValue: e.target.value
        });
    };

    handlePasswordConfirmChange = e => {
        this.setState({
            passwordConfirmValue: e.target.value
        });
    };

    handleKeyDownSubmit = e => {
        if (e.keyCode === 13) {
            this.handleClick();
        }
    };

    handleClick = () => {
        if (this.state.passwordValue !== this.state.passwordConfirmValue) {
            this.setState({
                errorMessage: getTranslation("login.errorMessage.bothPassText")
            });
            return;
        }

        authService.setInitialPassword(
            this.state.passwordValue,
            this.props.params.token,
            (err, response) => {
                if (err) {
                    console.log(err);
                } else {
                    if (response.success) {
                        this.setState({ done: true, errorMessage: "" });
                        window.setTimeout(() => browserHistory.push("/login"), 2000);
                    } else {
                        this.setState({
                            errorMessage: response.reason.message
                        });
                    }

                }
            }
        );
    };

    render() {
        const { passwordValue, passwordConfirmValue } = this.state;

        const validPassword =
            passwordValue.length >= 5 && passwordConfirmValue === passwordValue;

        const inputStyles = {
            width: "calc(100% - 4rem)"
        };

        const textStyles = {
            paddingBottom: "8px",
            fontSize: 16,
            color: "white"
        };

        const floatingLabelStyle = {
            color: "rgba(255, 255, 255, 0.3)"
        };

        const buttonStyles = {
            height: "50px"
        };

        const labelStyles = {
            fontSize: "16px",
            fontWeight: "bold",
            lineHeight: "50px"
        };

        const underlineStyles = {
            borderColor: "rgba(255,255,255, 0.1)",
            backgroundColor: "#5ebef3"
        };

        return (
            <div className="reset-container">
                {this.state.verified &&
                    !this.state.done && (
                        <div>
                            <div className="fields-outer-wrapper">
                                <span className="text">
                                    <Translate value="login.setInitial.title" />
                                </span>
                                <div className="forgot fields-wrapper">
                                    <TextField
                                        type="password"
                                        ref="password"
                                        value={this.state.passwordValue}
                                        onChange={this.handlePasswordChange}
                                        onKeyDown={this.handleKeyDownSubmit}
                                        style={inputStyles}
                                        inputStyle={textStyles}
                                        underlineStyle={underlineStyles}
                                        underlineShow={false}
                                        hintStyle={floatingLabelStyle}
                                        hintText={getTranslation("login.hintText.newPass")}
                                        className="text-field"
                                        autoFocus={true}
                                    />
                                    <hr />
                                    <TextField
                                        type="password"
                                        ref="password"
                                        value={this.state.passwordConfirmValue}
                                        onChange={this.handlePasswordConfirmChange}
                                        onKeyDown={this.handleKeyDownSubmit}
                                        style={inputStyles}
                                        inputStyle={textStyles}
                                        underlineStyle={underlineStyles}
                                        underlineShow={false}
                                        hintStyle={floatingLabelStyle}
                                        hintText={getTranslation("login.hintText.confirmPass")}
                                        className="text-field"
                                    />
                                </div>
                            </div>
                            {this.state.errorMessage && (
                                <p className="error-message">{this.state.errorMessage}</p>
                            )}
                            <RaisedButton
                                label={getTranslation("login.setInitial.setPassBtn")}
                                disabled={!validPassword}
                                primary={true}
                                fullWidth={true}
                                style={buttonStyles}
                                labelStyle={labelStyles}
                                onClick={event => this.handleClick(event)}
                            />
                        </div>
                    )}
                {this.state.done && (
                    <div>
                        <div className="fields-outer-wrapper">
                            <span className="text">
                                <Translate value="login.setInitial.successMsg" />
                            </span>
                            <div className="forgot fields-wrapper" />
                        </div>
                    </div>
                )}
            </div>
        );
    }
}
