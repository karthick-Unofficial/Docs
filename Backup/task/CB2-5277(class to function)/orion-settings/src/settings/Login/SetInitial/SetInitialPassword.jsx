import React, { useEffect, useState, useRef } from "react";

import TextField from "material-ui/TextField";
import RaisedButton from "material-ui/RaisedButton";

import { browserHistory } from "react-router";

import { authService } from "client-app-core";
import { Translate, getTranslation } from "orion-components/i18n/I18nContainer";

const SetInitialPassword = ({ errorMessage, params }) => {
	const [passwordValue, setPasswordValue] = useState("");
	const [passwordConfirmValue, setPasswordConfirmValue] = useState("");
	const [errorMsg, setErrorMsg] = useState(errorMessage || null);
	const [verified, setVerified] = useState(false);
	const [done, setDone] = useState(false);
	const password = useRef(null);

	useEffect(() => {
		authService.verifySetInitialPassword(
			params.token,
			(err, response) => {
				if (err) {
					console.log(err);
					browserHistory.push("/login/invalid-reset");
				} else {
					setVerified(true);
				}
			}
		);
	}, []);

	useEffect(() => {
		setErrorMsg(errorMessage);
	}, [errorMessage]);

	const handlePasswordChange = e => {
		setPasswordValue(e.target.value);
	};

	const handlePasswordConfirmChange = e => {
		setPasswordConfirmValue(e.target.value);
	};

	const handleKeyDownSubmit = e => {
		if (e.keyCode === 13) {
			handleClick();
		}
	};

	const handleClick = () => {
		if (passwordValue !== passwordConfirmValue) {
			setErrorMsg(getTranslation("login.errorMessage.bothPassText"));
			return;
		}

		authService.setInitialPassword(
			passwordValue,
			params.token,
			(err, response) => {
				if (err) {
					console.log(err);
				} else {
					if (response.success) {
						setDone(true);
						setErrorMsg("");
						window.setTimeout(() => browserHistory.push("/login"), 2000);
					} else {
						setErrorMsg(response.reason.message);
					}

				}
			}
		);
	};

	const validPassword =
		passwordValue.length >= 5 && passwordConfirmValue === passwordValue;

	const inputStyles = {
		width: "calc(100% - 4rem)"
	};

	const textStyles = {
		paddingBottom: "8px",
		fontSize: 16,
		color: "white"
	};

	const floatingLabelStyle = {
		color: "rgba(255, 255, 255, 0.3)"
	};

	const buttonStyles = {
		height: "50px"
	};

	const labelStyles = {
		fontSize: "16px",
		fontWeight: "bold",
		lineHeight: "50px"
	};

	const underlineStyles = {
		borderColor: "rgba(255,255,255, 0.1)",
		backgroundColor: "#5ebef3"
	};

	return (
		<div className="reset-container">
			{verified &&
				!done && (
				<div>
					<div className="fields-outer-wrapper">
						<span className="text">
							<Translate value="login.setInitial.title" />
						</span>
						<div className="forgot fields-wrapper">
							<TextField
								type="password"
								ref={password}
								value={passwordValue}
								onChange={handlePasswordChange}
								onKeyDown={handleKeyDownSubmit}
								style={inputStyles}
								inputStyle={textStyles}
								underlineStyle={underlineStyles}
								underlineShow={false}
								hintStyle={floatingLabelStyle}
								hintText={getTranslation("login.hintText.newPass")}
								className="text-field"
								autoFocus={true}
							/>
							<hr />
							<TextField
								type="password"
								ref={password}
								value={passwordConfirmValue}
								onChange={handlePasswordConfirmChange}
								onKeyDown={handleKeyDownSubmit}
								style={inputStyles}
								inputStyle={textStyles}
								underlineStyle={underlineStyles}
								underlineShow={false}
								hintStyle={floatingLabelStyle}
								hintText={getTranslation("login.hintText.confirmPass")}
								className="text-field"
							/>
						</div>
					</div>
					{errorMsg && (
						<p className="error-message">{errorMsg}</p>
					)}
					<RaisedButton
						label={getTranslation("login.setInitial.setPassBtn")}
						disabled={!validPassword}
						primary={true}
						fullWidth={true}
						style={buttonStyles}
						labelStyle={labelStyles}
						onClick={event => handleClick(event)}
					/>
				</div>
			)}
			{done && (
				<div>
					<div className="fields-outer-wrapper">
						<span className="text">
							<Translate value="login.setInitial.successMsg" />
						</span>
						<div className="forgot fields-wrapper" />
					</div>
				</div>
			)}
		</div>
	);
};

export default SetInitialPassword;