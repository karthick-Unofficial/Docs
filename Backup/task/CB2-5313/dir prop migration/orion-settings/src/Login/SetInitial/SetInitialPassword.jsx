import React, { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import { TextField, Button } from "@mui/material";


import { useNavigate } from "react-router-dom";

import { authService } from "client-app-core";
import { Translate, getTranslation } from "orion-components/i18n";

const SetInitialPassword = ({ errorMessage }) => {

	const navigate = useNavigate();
	const { token } = useParams();

	const [passwordValue, setPasswordValue] = useState("");
	const [passwordConfirmValue, setPasswordConfirmValue] = useState("");
	const [errorMsg, setErrorMsg] = useState(errorMessage || null);
	const [verified, setVerified] = useState(false);
	const [done, setDone] = useState(false);

	useEffect(() => {
		authService.verifySetInitialPassword(
			token,
			(err, response) => {
				if (err) {
					console.log(err);
					navigate("/login/invalid-reset");
				} else {
					setVerified(true);
				}
			}
		);
	}, []);

	useEffect(() => {
		if (errorMessage) {
			setErrorMsg(errorMessage);
		}
	}, [errorMessage]);

	const handlePasswordChange = e => {
		setPasswordValue(e.target.value);
	};

	const handlePasswordConfirmChange = e => {
		setPasswordConfirmValue(e.target.value);
	};

	const handleKeyDownSubmit = e => {
		if (e.keyCode === 13) {
			handleClick();
		}
	};

	const handleClick = () => {
		if (passwordValue !== passwordConfirmValue) {
			setErrorMsg(getTranslation("login.errorMessage.bothPassText"));
			return;
		}

		authService.setInitialPassword(
			passwordValue,
			token,
			(err, response) => {
				if (err) {
					console.log(err);
				} else {
					if (response.success) {
						setDone(true);
						setErrorMsg("");
						window.setTimeout(() => navigate("/login"), 2000);
					} else {
						setErrorMsg(response.reason.message);
					}

				}
			}
		);
	};

	const validPassword =
		passwordValue.length >= 5 && passwordConfirmValue === passwordValue;

	const inputStyles = {
		width: "calc(100% - 4rem)"
	};

	const textStyles = {
		paddingBottom: "8px",
		fontSize: 16,
		color: "white"
	};

	const floatingLabelStyle = {
		color: "rgba(255, 255, 255, 0.3)"
	};

	const buttonStyles = {
		height: "50px"
	};

	const labelStyles = {
		fontSize: "16px",
		fontWeight: "bold",
		lineHeight: "50px"
	};

	const underlineStyles = {
		borderColor: "rgba(255,255,255, 0.1)",
		backgroundColor: "#5ebef3"
	};

	return (
		<div className="reset-container">
			{verified &&
				!done && (
					<div>
						<div className="fields-outer-wrapper">
							<span className="text">
								<Translate value="login.setInitial.title" />
							</span>
							<div className="forgot fields-wrapper">
								<TextField
									variant="standard"
									type="password"
									value={passwordValue}
									onChange={handlePasswordChange}
									onKeyDown={handleKeyDownSubmit}
									style={inputStyles}
									inputStyle={textStyles}
									underlineStyle={underlineStyles}
									underlineShow={false}
									hintStyle={floatingLabelStyle}
									hintText={getTranslation("login.hintText.newPass")}
									className="text-field"
									autoFocus={true}
								/>
								<hr />
								<TextField
									variant="standard"
									type="password"
									value={passwordConfirmValue}
									onChange={handlePasswordConfirmChange}
									onKeyDown={handleKeyDownSubmit}
									style={inputStyles}
									inputStyle={textStyles}
									underlineStyle={underlineStyles}
									underlineShow={false}
									hintStyle={floatingLabelStyle}
									hintText={getTranslation("login.hintText.confirmPass")}
									className="text-field"
								/>
							</div>
						</div>
						{errorMsg && (
							<p className="error-message">{errorMsg}</p>
						)}
						<Button
							disabled={!validPassword}
							fullWidth={true}
							style={buttonStyles}
							onClick={event => handleClick(event)}
						>
							<div style={labelStyles}>
								{getTranslation("login.setInitial.setPassBtn")}
							</div>
						</Button>
					</div>
				)}
			{done && (
				<div>
					<div className="fields-outer-wrapper">
						<span className="text">
							<Translate value="login.setInitial.successMsg" />
						</span>
						<div className="forgot fields-wrapper" />
					</div>
				</div>
			)}
		</div>
	);
};

export default SetInitialPassword;