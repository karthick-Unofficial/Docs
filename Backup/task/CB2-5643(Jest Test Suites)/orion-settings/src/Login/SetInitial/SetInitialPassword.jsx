import React, { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import { Input, Button } from "@mui/material";


import { useNavigate } from "react-router-dom";

import { authService } from "client-app-core";
import { Translate, getTranslation } from "orion-components/i18n";
import { useSelector } from "react-redux";

const SetInitialPassword = ({ errorMessage }) => {

	const navigate = useNavigate();
	const { token } = useParams();

	// helps to render latest translations
	const translations = useSelector(state => state.i18n.translations);

	const [passwordValue, setPasswordValue] = useState("");
	const [passwordConfirmValue, setPasswordConfirmValue] = useState("");
	const [errorMsg, setErrorMsg] = useState(errorMessage || null);
	const [verified, setVerified] = useState(false);
	const [done, setDone] = useState(false);

	useEffect(() => {
		authService.verifySetInitialPassword(
			token,
			(err, response) => {
				if (err) {
					console.log(err);
					navigate("/login/invalid-reset");
				} else {
					setVerified(true);
				}
			}
		);
	}, []);

	useEffect(() => {
		if (errorMessage) {
			setErrorMsg(errorMessage);
		}
	}, [errorMessage]);

	const handlePasswordChange = e => {
		setPasswordValue(e.target.value);
	};

	const handlePasswordConfirmChange = e => {
		setPasswordConfirmValue(e.target.value);
	};

	const handleKeyDownSubmit = e => {
		if (e.keyCode === 13) {
			handleClick();
		}
	};

	const handleClick = () => {
		if (passwordValue !== passwordConfirmValue) {
			setErrorMsg(getTranslation("login.errorMessage.bothPassText"));
			return;
		}

		authService.setInitialPassword(
			passwordValue,
			token,
			(err, response) => {
				if (err) {
					console.log(err);
				} else {
					if (response.success) {
						setDone(true);
						setErrorMsg("");
						window.setTimeout(() => navigate("/login"), 2000);
					} else {
						setErrorMsg(response.reason.message);
					}

				}
			}
		);
	};

	const validPassword =
		passwordValue.length >= 5 && passwordConfirmValue === passwordValue;

	const inputStyles = {
		width: "calc(100% - 4rem)",
		paddingTop: "10px",
		paddingBottom: "15px",
		fontSize: 16,
		color: "white"
	};

	const buttonStyles = {
		height: "50px",
		color: "#fff",
		fontSize: "16px",
		fontWeight: "bold",
		lineHeight: "50px"
	};

	const labelStyles = {
		fontSize: "16px",
		fontWeight: "bold",
		lineHeight: "50px"
	};

	return (
		<div className="reset-container">
			{verified &&
				!done && (
				<div>
					<div className="fields-outer-wrapper">
						<span className="text">
							<Translate value="login.setInitial.title" />
						</span>
						<div className="forgot fields-wrapper">
							<Input
								id="newPass"
								type="password"
								placeholder={getTranslation("login.hintText.newPass")}
								autoFocus={true}
								disableUnderline={true}
								onChange={handlePasswordChange}
								onKeyDown={handleKeyDownSubmit}
								value={passwordValue}
								style={inputStyles}
								className="text-field"

							/>
							<hr />
							<Input
								id="confirmPass"
								type="password"
								placeholder={getTranslation("login.hintText.confirmPass")}
								autoFocus={true}
								disableUnderline={true}
								onChange={handlePasswordConfirmChange}
								onKeyDown={handleKeyDownSubmit}
								value={passwordConfirmValue}
								style={inputStyles}
								className="text-field"
							/>
						</div>
					</div>
					{errorMsg && (
						<p className="error-message">{errorMsg}</p>
					)}
					<Button
						disabled={!validPassword}
						fullWidth={true}
						style={buttonStyles}
						onClick={event => handleClick(event)}
					>
						<div style={labelStyles}>
							{getTranslation("login.setInitial.setPassBtn")}
						</div>
					</Button>
				</div>
			)}
			{done && (
				<div>
					<div className="fields-outer-wrapper">
						<span className="text">
							<Translate value="login.setInitial.successMsg" />
						</span>
						<div className="forgot fields-wrapper" />
					</div>
				</div>
			)}
		</div>
	);
};

export default SetInitialPassword;