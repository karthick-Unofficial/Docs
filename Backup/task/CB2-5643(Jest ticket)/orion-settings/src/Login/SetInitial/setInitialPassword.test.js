import React from "react";
import SetInitialPassword from "./SetInitialPassword";

import { shallow } from "enzyme";

jest.mock('react-redux', () => ({
	useSelector: jest.fn().mockImplementation(selector => selector()),
	useDispatch: () => jest.fn()
}));

jest.mock('react-router-dom', () => ({
	useNavigate: () => jest.fn(),
	useParams: () => jest.fn()
}));

describe("setInitialPassword", () => {
	it("renders", () => {
		const wrapper = shallow(
			<SetInitialPassword errorMessage={""} />
		);
		expect(wrapper).toMatchSnapshot();
	});
});