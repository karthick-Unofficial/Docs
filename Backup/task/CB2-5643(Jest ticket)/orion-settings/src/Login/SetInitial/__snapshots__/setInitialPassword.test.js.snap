// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setInitialPassword renders 1`] = `ShallowWrapper {}`;
